.github/workflows/update-flake.yml

```diff
@@ -0,0 +1,27 @@
+name: Update Nix flake.lock
+on:
+  workflow_dispatch:
+  schedule:
+    # weekly on Mondays at 3:17 UTC
+    - cron: "17 3 * * 1"
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  flake-lock-file:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+        with:
+          fetch-depth: 0
+      - uses: DeterminateSystems/nix-installer-action@e50d5f73bfe71c2dd0aa4218de8f4afa59f8f81d
+      - uses: DeterminateSystems/magic-nix-cache-action@87b14cf437d03d37989d87f0fa5ce4f5dc1a330b
```

Comment:
These specific revs were pulled across from the [build-nix workflow](https://github.com/shanesveller/jj/blob/32d2a85539254e9d96f9819072fa5c6ac70dd1e4/.github/workflows/build-nix.yml#L26-L30).